### PR TITLE
Removed offending tr1 references.

### PR DESCRIPTION
**Pull request by bearbin**
This should fix the mac build.

### CODE REVIEW

File: Tools/ProtoProxy/Globals.h [hunk 1]
```diff
@@ -223,13 +220,9 @@ typedef unsigned char Byte;
 #if (defined(_MSC_VER) && (_MSC_VER < 1600))
 	// MSVC before 2010 doesn't have std::shared_ptr, but has std::tr1::shared_ptr, defined in <memory> included earlier
 	#define SharedPtr std::tr1::shared_ptr
```
Comment (worktycho) [inline Tools/ProtoProxy/Globals.h]: Remove this, we don't build on MSVC < 2010 any more

File: Tools/QtBiomeVisualiser/Globals.h [hunk 2]
```diff
@@ -335,13 +335,9 @@ void inline LOGERROR(const char* a_Format, ...)
 #if (defined(_MSC_VER) && (_MSC_VER < 1600))
 	// MSVC before 2010 doesn't have std::shared_ptr, but has std::tr1::shared_ptr, defined in <memory> included earlier
 	#define SharedPtr std::tr1::shared_ptr
```
Comment (worktycho) [inline Tools/QtBiomeVisualiser/Globals.h]: Remove this

File: Tools/ToLuaDoxy/Globals.h [hunk 3]
```diff
@@ -43,10 +43,6 @@
 	#include <semaphore.h>
 	#include <errno.h>
 	#include <fcntl.h>
-	
-	#if !defined(ANDROID_NDK)
-		#include <tr1/memory>
```
Comment (worktycho) [inline Tools/ToLuaDoxy/Globals.h]: We need <memory> now.

File: Tools/ProtoProxy/Globals.h [hunk 4]
```diff
@@ -12,26 +12,26 @@
...
 	#define FORMATSTRING(formatIndex, va_argsIndex)
 
 #elif defined(__GNUC__)
 
 	// TODO: Can GCC explicitly mark classes as abstract (no instances can be created)?
 	#define abstract
-	
+
 	// TODO: Can GCC mark virtual methods as overriding (forcing them to have a virtual function of the same signature in the base class)
 	#define override
```
Comment (SphinxC0re) [inline Tools/ProtoProxy/Globals.h]: isn't gcc totally capable of interpreting the override symbol?

Comment (worktycho) [inline Tools/ProtoProxy/Globals.h]: Thats ancient. I think it might cause an error in clang now to define this macro.

Comment (SphinxC0re) [inline Tools/ProtoProxy/Globals.h]: It's even in the standard: http://en.cppreference.com/w/cpp/language/override

File: Tools/ToLuaDoxy/Globals.h (hunk 3 above)
Comment (bearbin) [inline Tools/ToLuaDoxy/Globals.h]: Memory is already included in that file, does it need to be included again?

Comment (worktycho) [inline Tools/ToLuaDoxy/Globals.h]: No. I just can see it.

**Comment by bearbin** [comment]
The mac build is confirmed fixed.

**Pull request merged**